src/test/scala/io/iohk/ethereum/forkid/ForkIdValidatorSpec.scala

```diff
@@ -0,0 +1,102 @@
+package io.iohk.ethereum.forkid
+
+import akka.util.ByteString
+import io.iohk.ethereum.forkid.ForkId._
+import io.iohk.ethereum.utils.Config._
+import io.iohk.ethereum.utils.ForkBlockNumbers
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.bouncycastle.util.encoders.Hex
+import org.scalatest.matchers.should._
+import org.scalatest.wordspec.AnyWordSpec
+
+import scala.concurrent.duration._
+
+import ForkIdValidator._
+
+class ForkIdValidatorSpec extends AnyWordSpec with Matchers {
+
+  val config = blockchains
+
+  val ethGenesisHash = ByteString(Hex.decode("d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3"))
+
+  "ForkIdValidator" must {
+    "correctly validate ETH peers" in {
+      // latest fork at the time of writing those assertions (in the spec) was Petersburg
+      val ethForksList: List[BigInt] = List(1150000, 1920000, 2463000, 2675000, 4370000, 7280000)
```

Comment:
Are all these different forks interesting for the purposes of testing? I mean, you are testing it with 6, would it make a material difference to the test quality if it was 5 or 7 or 9000?

Also, would this be a good opportunity for property based testing perhaps?

Reply:
now I'm curious - how does one property test an equation? by writing the same equation in a different way?

Reply:
Yes, but part of this 'equation' would be in how the arbitrary data is constructed. I agree that just re-implementing the same logic wouldn't make much sense.

Reply:
oh this isn't arbitrary data - it's hard fork heights.

Reply:
This test was provided as part of the specification in https://eips.ethereum.org/EIPS/eip-2124 so we would have to have a really good reason to modify or not include it here. 

Whether all the forks are interesting for the purposes of testing is probably a question to the original EIP's authors. 

@jvdp what are the properties you would like to test here?